pkg/nrtupdater/node_getter.go

```diff
@@ -34,27 +42,34 @@ func (ng *DisabledNodeGetter) Get(ctx context.Context, nodeName string, opts met
 	return nil, fmt.Errorf("%w", NotConfigured)
 }
 
-type OneshotNodeGetter struct {
-	K8SCli kubernetes.Interface
-	node   *corev1.Node
+type CachedNodeGetter struct {
+	nodes map[string]*corev1.Node
 }
 
-func (ng *OneshotNodeGetter) Get(ctx context.Context, nodeName string, opts metav1.GetOptions) (*corev1.Node, error) {
-	if ng.node != nil {
-		return ng.node, nil
+func NewCachedNodeGetter(k8sInterface kubernetes.Interface, ctx context.Context) *CachedNodeGetter {
+	nodelist, err := k8sInterface.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil
+	}
+
+	retVal := &CachedNodeGetter{nodes: make(map[string]*corev1.Node, len(nodelist.Items))}
+	for _, node := range nodelist.Items {
+		retVal.nodes[node.Name] = &node
```

Comment:
this is dangerous and could hit a common go gotcha: https://github.com/golang/go/wiki/CommonMistakes

the fix is simple:
```
	for idx := range nodelist.Items {
                node := &nodelist.Items[idx]
		retVal.nodes[node.Name] = node
```

Reply:
Thanks for the catch and the link.